libraries/SITL/SIM_Vicon.cpp

```diff
@@ -204,6 +204,45 @@ void Vicon::update_vicon_position_estimate(const Location &loc,
             NULL, 0);
         msg_buf[msg_buf_index].time_send_us = time_send_us;
     }
+
+    // determine time, position, and angular deltas
+    static uint64_t prev_time_us = now_us + time_offset_us;
+    static Vector3f prev_position(pos_corrected.x, pos_corrected.y, pos_corrected.z);
+
+    Quaternion curr_quat;
+    curr_quat.from_euler(roll, pitch, yaw);
+    static Quaternion prev_quat(curr_quat);
+
+    curr_quat = curr_quat.angular_difference(prev_quat);
+
+    float angle_delta[3] = {curr_quat.get_euler_roll(),
+                                curr_quat.get_euler_pitch(),
+                                wrap_PI(curr_quat.get_euler_yaw())};
+
+    float postion_delta[3] = {pos_corrected.x - prev_position.x,
+                                pos_corrected.y - prev_position.y,
+                                pos_corrected.z - prev_position.z};
+
+    // send vision position delta
+    if (should_send(ViconTypeMask::VISION_POSITION_DELTA) && get_free_msg_buf_index(msg_buf_index)) {
+        mavlink_msg_vision_position_delta_pack_chan(
+            system_id,
+            component_id,
+            mavlink_ch,
+            &msg_buf[msg_buf_index].obs_msg,
+            now_us + time_offset_us,
+            (prev_time_us - now_us + time_offset_us),
+            angle_delta,
+            postion_delta,
+            NAN);
```

Comment:
this needs to be a number from 0 to 100 or else it won't be consumed by the EKF.  We might as well make it 100 I think.

Reply:
The EKF scales the velocity_error  from min. to a max. See this line: [https://github.com/ArduPilot/ardupilot/blob/be8f747b7a471b99336c8e49ce0393710e3fe162/libraries/AP_NavEKF3/AP_NavEKF3_Measurements.cpp#L1399](url)
Plus the other messages here do not set confidence/covariance yet.